### PR TITLE
Figcaption as a tag for Text

### PR DESCRIPTION
**Pull request by kof**
## Description

Closes https://github.com/webstudio-is/webstudio-builder/issues/1377

## Steps for reproduction

1. add Text
2. change tag prop to figcaption

## Code Review

- [ ] hi @TrySound , I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: packages/react-sdk/src/components/text-block.tsx [hunk 1]
```diff
@@ -1,10 +1,19 @@
...
 
 export const defaultTag = "div";
 
-export const TextBlock = forwardRef<
-  ElementRef<typeof defaultTag>,
-  ComponentProps<typeof defaultTag>
->((props, ref) => <div {...props} ref={ref} />);
+// We don't want to enable all tags because Box is usually a container and we have specific components for many tags.
+type Props = ComponentProps<typeof defaultTag> & {
+  tag?: "div" | "span" | "figcaption";
```
Comment (TrySound) [inline packages/react-sdk/src/components/text-block.tsx:12]: Span is not block

Comment (kof) [reply packages/react-sdk/src/components/text-block.tsx:12]: Yes, which is why we need to rename to Text, we already discussed it in another PR

Comment (kof) [reply packages/react-sdk/src/components/text-block.tsx:12]: Sometimes you need a span

Comment (kof) [reply packages/react-sdk/src/components/text-block.tsx:12]: e.g. when text is inside a label, you don't want a div https://github.com/webstudio-is/webstudio-builder/issues/1552

Comment (TrySound) [reply packages/react-sdk/src/components/text-block.tsx:12]: Label is rich text though

File: packages/react-sdk/src/components/text-block.tsx [hunk 2]
```diff
@@ -1,10 +1,19 @@
...
+};
 
-TextBlock.displayName = "TextBlock";
+export const TextBlock = forwardRef<ElementRef<typeof defaultTag>, Props>(
+  ({ tag = defaultTag, ...props }, ref) => {
+    return createElement(tag as string, { ...props, ref });
```
Comment (TrySound) [inline packages/react-sdk/src/components/text-block.tsx:17]: Why as string?

Comment (TrySound) [reply packages/react-sdk/src/components/text-block.tsx:17]: Seems like box works without it

Comment (kof) [reply packages/react-sdk/src/components/text-block.tsx:17]: I think it used to not work without it, removing from both

File: packages/react-sdk/src/components/text-block.tsx (hunk 1 above)
Comment (kof) [reply packages/react-sdk/src/components/text-block.tsx:12]: Let's discuss in dm if its a problem, merging now

**Pull request merged**